tokio-postgres/src/client.rs

```diff
@@ -226,6 +226,25 @@ impl Client {
             .await
     }
 
+    /// Same as `query` but returns text results instead of binary
+    ///
+    /// We introduce an additional method rather than parameters not to break all
+    /// the conversions and marshalling supplied by `postgres-types` and feature crates and
+    /// inasmuch as this allows us to maintain API compatibility with upstream
+    pub async fn query_with_text_results<T>(
+        &self,
+        statement: &T,
+        params: &[&(dyn ToSql + Sync)],
+    ) -> Result<Vec<Row>, Error>
```

Comment:
This shouldn't use the `Row` type since `FromSql` will not work with almost any of the types returned in text mode.